File: VB/19_OLE/EmbedExcelAsOLE.vb
Imports Spire.Presentation
Imports Spire.Presentation.Drawing
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.IO
Imports System.Text
Imports System.Windows.Forms

Namespace EmbedExcelAsOLE
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs)
			'Create a Presentaion document
			Dim ppt As New Presentation()

			'Load a image file
			Dim image As Image = Image.FromFile("..\..\..\..\..\..\Data\EmbedExcelAsOLE.png")
			Dim oleImage As IImageData = ppt.Images.Append(image)

			'////////////////Use the following code for netstandard dlls///////////////////////// 
'            
'            FileStream fileStream = new FileStream(@"..\..\..\..\..\..\Data\EmbedExcelAsOLE.png", FileMode.Open, FileAccess.Read, FileShare.Read);
'            byte[] bytes = new byte[fileStream.Length];
'            fileStream.Read(bytes, 0, bytes.Length);
'            fileStream.Close();
'            Stream stream = new MemoryStream(bytes);          
'            IImageData oleImage = ppt.Images.Append(stream);
'            stream.Close();
'            fileStream.Close();
'            SkiaSharp.SKBitmap image = SkiaSharp.SKBitmap.Decode(@"..\..\..\..\..\..\Data\EmbedExcelAsOLE.png");
'            

			Dim rec As New Rectangle(80, 60, image.Width, image.Height)

			'Insert an OLE object to presentation based on the Excel data
			Dim oleObject As Spire.Presentation.IOleObject = ppt.Slides(0).Shapes.AppendOleObject("excel", File.ReadAllBytes("..\..\..\..\..\..\Data\EmbedExcelAsOLE.xlsx"), rec)
			oleObject.SubstituteImagePictureFillFormat.Picture.EmbedImage = oleImage
			oleObject.ProgId = "Excel.Sheet.12"

			'Save the document
			ppt.SaveToFile("EmbedExcelAsOLE_result.pptx", Spire.Presentation.FileFormat.Pptx2010)
			System.Diagnostics.Process.Start("EmbedExcelAsOLE_result.pptx")
		End Sub
	End Class
End Namespace

File: CS/19_OLE/EmbedExcelAsOLE.cs
using Spire.Presentation;
using Spire.Presentation.Drawing;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Text;
using System.Windows.Forms;

namespace EmbedExcelAsOLE
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
			//Create a Presentaion document
			Presentation ppt = new Presentation();
			
			//Load a image file
            Image image = Image.FromFile(@"..\..\..\..\..\..\Data\EmbedExcelAsOLE.png");
            IImageData oleImage = ppt.Images.Append(image);
           
            //////////////////Use the following code for netstandard dlls///////////////////////// 
            /*
            FileStream fileStream = new FileStream(@"..\..\..\..\..\..\Data\EmbedExcelAsOLE.png", FileMode.Open, FileAccess.Read, FileShare.Read);
            byte[] bytes = new byte[fileStream.Length];
            fileStream.Read(bytes, 0, bytes.Length);
            fileStream.Close();
            Stream stream = new MemoryStream(bytes);          
            IImageData oleImage = ppt.Images.Append(stream);
            stream.Close();
            fileStream.Close();
            SkiaSharp.SKBitmap image = SkiaSharp.SKBitmap.Decode(@"..\..\..\..\..\..\Data\EmbedExcelAsOLE.png");
            */
            
            Rectangle rec = new Rectangle(80, 60, image.Width, image.Height);

            //Insert an OLE object to presentation based on the Excel data
            Spire.Presentation.IOleObject oleObject = ppt.Slides[0].Shapes.AppendOleObject("excel", File.ReadAllBytes(@"..\..\..\..\..\..\Data\EmbedExcelAsOLE.xlsx"), rec);
            oleObject.SubstituteImagePictureFillFormat.Picture.EmbedImage = oleImage;
            oleObject.ProgId = "Excel.Sheet.12";

            //Save the document
            ppt.SaveToFile("EmbedExcelAsOLE_result.pptx", Spire.Presentation.FileFormat.Pptx2010);
            System.Diagnostics.Process.Start("EmbedExcelAsOLE_result.pptx");
        }
    }
}
